lib/host/wasi/wasifunc.cpp

```diff
@@ -1525,6 +1548,164 @@ Expect<uint32_t> WasiSchedYield::body(Runtime::Instance::MemoryInstance *) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+Expect<uint32_t> WasiSockOpen::body(Runtime::Instance::MemoryInstance *MemInst,
+                                    int32_t AddressFamily, int32_t SockType,
+                                    uint32_t RoFdPtr) {
+  /// Check memory instance from module.
+  if (MemInst == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+
+  __wasi_fd_t *const RoFd =
+      MemInst->getPointer<__wasi_fd_t *>(RoFdPtr, sizeof(__wasi_fd_t));
+  if (RoFd == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+
+  __wasi_address_family_t WasiAddressFamily;
+  if (auto Res = cast<__wasi_address_family_t>(AddressFamily); unlikely(!Res)) {
+    return Res.error();
+  } else {
+    WasiAddressFamily = *Res;
+  }
+
+  __wasi_sock_type_t WasiSockType;
+  if (auto Res = cast<__wasi_sock_type_t>(SockType); unlikely(!Res)) {
+    return Res.error();
+  } else {
+    WasiSockType = *Res;
+  }
+
+  if (auto Res = Env.sockOpen(WasiAddressFamily, WasiSockType);
+      unlikely(!Res)) {
+    return Res.error();
+  } else {
+    *RoFd = *Res;
+  }
+
+  return __WASI_ERRNO_SUCCESS;
+}
+
+Expect<uint32_t> WasiSockBind::body(Runtime::Instance::MemoryInstance *MemInst,
+                                    int32_t Fd, uint32_t AddressPtr,
+                                    uint32_t Port) {
+  /// Check memory instance from module.
+  if (MemInst == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  if (Fd < 0) {
+    return __WASI_ERRNO_BADF;
+  }
+
+  __wasi_address_t *InnerAddress = MemInst->getPointer<__wasi_address_t *>(
+      AddressPtr, sizeof(__wasi_address_t));
+  if (InnerAddress == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+
+  if (InnerAddress->buf_len != 4 && InnerAddress->buf_len != 16) {
+    return __WASI_ERRNO_INVAL;
+  }
+
+  uint8_t *AddressBuf = MemInst->getPointer<uint8_t *>(
+      InnerAddress->buf, sizeof(uint8_t) * InnerAddress->buf_len);
+  if (AddressBuf == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  const __wasi_fd_t WasiFd = Fd;
+
+  if (auto Res = Env.sockBind(WasiFd, AddressBuf, InnerAddress->buf_len,
```

Comment:
Because the FD will be bypassed into `Env.sockBind`, you just need to check its value inside. No need to do in this function. Same as the other functions in this file.